react/components/SelectionListOrderBy.js

```diff
@@ -1,43 +1,28 @@
-import React, { Component } from 'react'
+import React, { useState, useCallback } from 'react'
 import PropTypes from 'prop-types'
 import { injectIntl, intlShape } from 'react-intl'
 import classNames from 'classnames'
 import { find, propEq } from 'ramda'
 
-import { withRuntimeContext, Link } from 'vtex.render-runtime'
+import OutsideClickHandler from 'react-outside-click-handler'
+import { useRuntime, Link } from 'vtex.render-runtime'
 import { IconCaret } from 'vtex.dreamstore-icons'
 
 import searchResult from '../searchResult.css'
 
-class SelectionListOrderBy extends Component {
-  state = {
-    showDropdown: false,
-  }
+const SelectionListOrderBy = ({ orderBy, getLinkProps, options, intl }) => {
+  const [showDropdown, setShowDropdown] = useState(false)
 
-  handleDropdownBtClick = e => {
-    e.preventDefault()
-    this.setState({ showDropdown: !this.state.showDropdown })
-  }
+  const handleDropdownBtClick = useCallback(
+    () => setShowDropdown(!showDropdown),
+    [showDropdown]
+  )
 
-  handleOutsideClick = e => {
-    this.setState({ showDropdown: false })
-  }
+  const handleOutsideClick = useCallback(() => setShowDropdown(false), [
+    showDropdown,
```

Comment:
you don't need `showDropdown` here. please, read the [`useCallback` docs](https://reactjs.org/docs/hooks-reference.html#usecallback)